src/wallet/wallet.cpp

```diff
@@ -3295,6 +3311,9 @@ bool CWallet::TopUpKeyPool(unsigned int kpSize)
         if (IsLocked())
             return false;
 
+        // If wallet supports HD but no seed is present, do not top up keypool
+        if (IsBlank()) return true;
```

Comment:
In commit "[wallet] support creating a blank wallet" (96476657a06797913edffd42d3fdae320b86e9df):

This isn't checking the WALLET_FLAG_BLANK_WALLET flag at all. It seems weird that to allow topping up keypool or generating keys (in GetKeyFromPool) in a wallet that supports hd but intentionally has no seed. Am I understanding this correctly, or is there another check somewhere else?

Reply:
I can't follow your comment.
The `IsBlank()` check here stops the refill-process if the wallet is either blank or if no HD seed is available (while HD is enabled).
I also tested this by creating a blank wallet, import a watch-only and then calling `keypoolrefill`.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/15226#discussion_r254514195

> I can't follow your comment.

This comment was wrong about the keypool, but it was added reviewing a previous version of this PR (commit hash mentioned was 96476657a06797913edffd42d3fdae320b86e9df) before isBlank checked the blank flag, which is probably why you were confused.

I actually think now that this shouldn't directly be checking the blank flag, but I want to look more closely and follow up with an other review.